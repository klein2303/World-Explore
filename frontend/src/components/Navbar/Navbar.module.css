.navbar {
    background-color: white;
    width: 100%;
    height: 120px;
    position: relative;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    padding-top: 20px;
    font-family: "Poppins", sans-serif;
}

.navTitle,
.navTitle:hover,
.navTitle:focus {
    height: max-content;
    width: 15%;
    margin-left: 40px;
    text-decoration: none;
    color: #424242;
    font-size: 30px;
}

.navlinks {
    display: flex;
    flex-direction: row;
    gap: 10px;
    justify-content: end;
    align-items: center;
    width: 85%;
    margin-right: 40px;
}

.navlink,
.navlink:hover,
.navlink:focus {
    text-decoration: none;
    color: #424242;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    margin: 10px;
    font-size: 18px;
}

.profile {
    font-size: 40px;
    font-weight: lighter;
    color: #424242;
    padding-left: 20px;
}

.hamburgmenu {
    display: none;
}

.signout {
    margin-top: 40px;
    background-color: #e4e1f7;
    padding: 10px;
    border-radius: 10px;
}

.profilemenu {
    background-color: white;
    display: flex;
    flex-direction: column;
    justify-content: end;
    text-align: end;
    align-items: flex-end;
    position: absolute;
    right: 0px;
    top: 38px;
    margin-right: 40px;
    height: fit-content;
    width: 125px;
    z-index: 50;

    font-family: "Poppins", sans-serif;
}

.crosspos {
    display: flex;
    justify-content: end;
    width: 100%;
    padding-top: 22px;
    padding-bottom: 20px;
}

.cross {
    font-size: 1.6em;
    color: #424242;
    cursor: pointer;
}

@media (max-width: 800px) {
    .hamburgmenu {
        display: flex;
        justify-content: start;
        align-items: center;
        font-size: 1.6em;
        color: #424242;
        width: 65px;
        cursor: pointer;
        padding-left: 10px;
    }

    .navTitle {
        display: none;
    }

    .navlinks {
        display: none;
    }

    .crosspos {
        display: flex;
        justify-content: start;
        width: 100%;
        padding-top: 22px;
        padding-bottom: 20px;
    }

    .cross {
        font-size: 1.6em;
        color: #424242;
        cursor: pointer;
    }

    .navlink {
        width: 100%;
        height: 20px;
    }

    .navbar {
        background-color: white;
        width: 100%;
        height: 70px;
        position: relative;
        gap: 20%;
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: flex-start;
    }

    .mobilemenu {
        background-color: white;
        display: flex;
        flex-direction: column;
        justify-content: start;
        text-align: start;
        align-items: baseline;
        position: absolute;
        left: 0;
        top: 20px;
        margin-left: 20px;
        height: fit-content;
        width: 125px;
        z-index: 50;

        font-family: "Poppins", sans-serif;
    }
}
