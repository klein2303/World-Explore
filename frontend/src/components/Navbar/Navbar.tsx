import { Link } from "react-router-dom";
import styles from "./Navbar.module.css";
import { RxHamburgerMenu, RxCross1 } from "react-icons/rx";
import { useState } from "react";
import { CgProfile } from "react-icons/cg";
import { GoSearch } from "react-icons/go";
import { LuMapPin } from "react-icons/lu";

const Navbar = () => {
    const [isOpen, setIsOpen] = useState<boolean>(false);
    const [isProfileOpen, setIsProfileOpen] = useState<boolean>(false);

    const token = sessionStorage.getItem("auth-token");

    const handleSignOut = () => {
        sessionStorage.removeItem("auth-token");
        sessionStorage.removeItem("user");
    };

    return (
        <>
            {token ? (
                <main>
                    {/* Main navigation section of the page */}
                    <nav className={styles.navbar} role="navigation" aria-label="Main Navigation">
                        {isOpen && (
                            <div className={styles.mobilemenu} role="dialog" aria-label="Mobile Navigation Menu">
                                {/* Close button for mobile menu */}
                                <div className={styles.crosspos} role="presentation">
                                    <RxCross1
                                        className={styles.cross}
                                        onClick={() => setIsOpen(false)}
                                        aria-label="Close Mobile Menu"
                                    />
                                </div>

                                <Link to={"/"} className={styles.navlink} aria-label="Home">
                                    <p>Home</p>
                                </Link>
                                <Link
                                    to={"/ExploreCountries"}
                                    className={styles.navlink}
                                    aria-label="Explore Countries">
                                    <p>Explore Countries</p>
                                </Link>
                                <Link to={"/MyJournals"} className={styles.navlink} aria-label="My Journals">
                                    <p>My Journals</p>
                                </Link>

                                <Link to={"/"} className={styles.signOut} onClick={handleSignOut} aria-label="Sign out">
                                    <p>Sign Out</p>
                                </Link>
                            </div>
                        )}

                        {/* Button to open mobile menu */}
                        <RxHamburgerMenu
                            className={styles.hamburgmenu}
                            onClick={() => setIsOpen(true)}
                            aria-label="Open Mobile Menu"
                        />

                        <Link to={"/"} className={styles.navTitle}>
                            <p>WorldExplore</p>
                        </Link>

                        <div className={styles.navlinks} role="group" aria-label="Navigation Links">
                            <Link to={"/ExploreCountries"} className={styles.navlink}>
                                <GoSearch className={styles.search} aria-hidden="true" />
                                <p>Explore Countries</p>
                            </Link>
                            <Link to={"/MyJournals"} className={styles.navlink}>
                                <LuMapPin className={styles.pin} aria-hidden="true" />
                                <p>My Journals</p>
                            </Link>

                            {/* Profile button to open profile menu */}
                            <CgProfile
                                className={styles.profile}
                                onClick={() => setIsProfileOpen(true)}
                                aria-label="Open Profile Menu"
                            />

                            {/* Profile menu when open */}
                            {isProfileOpen && (
                                <div
                                    className={styles.profilemenu}
                                    role="dialog"
                                    aria-modal="true"
                                    aria-label="Profile Menu">
                                    {/* Close button for profile menu */}
                                    <div className={styles.crosspos} role="presentation">
                                        <CgProfile
                                            className={styles.profile}
                                            onClick={() => setIsProfileOpen(false)}
                                            aria-label="Close Profile Menu"
                                        />
                                    </div>

                                    <Link
                                        to={"/"}
                                        className={styles.signOut}
                                        onClick={handleSignOut}
                                        aria-label="Sign out">
                                        <p>Sign out</p>
                                    </Link>
                                </div>
                            )}
                        </div>
                    </nav>
                </main>
            ) : (
                <main>
                    <nav className={styles.notLoggedNavbar} role="navigation" aria-label="Main Navigation">
                        <Link to={"/Login"} className={styles.login}>
                            <p>Log in</p>
                        </Link>
                    </nav>
                </main>
            )}
        </>
    );
};

export default Navbar;
