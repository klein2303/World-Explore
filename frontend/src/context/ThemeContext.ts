import { createContext, useContext } from "react";

// Define the possible theme values
export type Theme = "light" | "dark";

// Define the shape of the theme context
export interface ThemeContextProps {
    theme: Theme;
    toggleTheme: () => void;
}

// Create the ThemeContext with an undefined default value
export const ThemeContext = createContext<ThemeContextProps | undefined>(undefined);

// Custom hook to use the ThemeContext
export const useTheme = (): ThemeContextProps => {
    const context = useContext(ThemeContext);
    if (!context) {
        throw new Error("useTheme must be used within a ThemeProvider");
    }
    return context;
};