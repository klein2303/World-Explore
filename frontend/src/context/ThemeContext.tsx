import React, { createContext, useState, useEffect, useContext } from 'react';

type Theme = 'light' | 'dark';

interface ThemeContextProps {
  theme: Theme;
  toggleTheme: () => void;
}

const ThemeContext = createContext<ThemeContextProps | undefined>(undefined);

export const ThemeProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [theme, setTheme] = useState<Theme>('light');

  // Load the theme from localStorage
  useEffect(() => {
    const savedTheme = localStorage.getItem('theme') as Theme;
    if (savedTheme) {
      setTheme(savedTheme);
    }
  }, []);

  // Apply the theme to the body element
  useEffect(() => {
    document.body.setAttribute('data-theme', theme);
    localStorage.setItem('theme', theme); // Save theme to localStorage for persistence
  }, [theme]);

  const toggleTheme = () => {
    setTheme(prevTheme => (prevTheme === 'light' ? 'dark' : 'light'));
  };

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

export const useTheme = (): ThemeContextProps => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
};
