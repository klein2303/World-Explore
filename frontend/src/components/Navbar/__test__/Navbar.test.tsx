import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { describe, it, expect, vi, beforeEach } from "vitest";
import Navbar from "../Navbar";
import { ThemeContext } from "../../../context/ThemeContext";

// Mock sessionStorage
const mockSessionStorage = (() => {
    let store: Record<string, string> = {};
    return {
        getItem: (key: string) => store[key] || null,
        setItem: (key: string, value: string) => (store[key] = value.toString()),
        removeItem: (key: string) => delete store[key],
        clear: () => (store = {}),
    };
})();
vi.stubGlobal("sessionStorage", mockSessionStorage);

// Mock `window.matchMedia`
vi.stubGlobal("matchMedia", (query: string) => ({
    matches: query === "(max-width: 800px)",
    media: query,
    addEventListener: vi.fn(),
    removeEventListener: vi.fn(),
}));

vi.mock("focus-trap-react", () => ({
    __esModule: true,
    default: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
}));

describe("Navbar Component", () => {
    const mockToggleTheme = vi.fn();
    const mockTheme = "light";

    beforeEach(() => {
        sessionStorage.clear();
    });

    const renderWithTheme = (ui: React.ReactElement) => {
        return render(
            <MemoryRouter>
                <ThemeContext.Provider value={{ theme: mockTheme, toggleTheme: mockToggleTheme }}>
                    {ui}
                </ThemeContext.Provider>
            </MemoryRouter>,
        );
    };

    it("renders correctly when user is logged in and matches snapshot", () => {
        sessionStorage.setItem("auth-token", "mock-token");

        const { asFragment } = renderWithTheme(<Navbar />);

        // Verify elements
        expect(screen.getByText("WorldExplore")).toBeInTheDocument();
        expect(screen.getByText("Explore Countries")).toBeInTheDocument();
        expect(screen.getByText("My Journals")).toBeInTheDocument();
        expect(screen.getByText("Sign Out")).toBeInTheDocument();

        // Match snapshot
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders correctly when user is not logged in and matches snapshot", () => {
        const { asFragment } = renderWithTheme(<Navbar />);

        // Verify elements
        expect(screen.getByText("WorldExplore")).toBeInTheDocument();
        expect(screen.getByText("Log in")).toBeInTheDocument();

        // Match snapshot
        expect(asFragment()).toMatchSnapshot();
    });

    it("opens and closes the mobile menu", async () => {
        sessionStorage.setItem("auth-token", "mock-token");

        renderWithTheme(<Navbar />);

        // Open mobile menu using the test ID
        const hamburgerButton = screen.getByTestId("HamburgerMenu");
        fireEvent.click(hamburgerButton);

        // Wait for the mobile menu to appear
        await waitFor(() => expect(screen.getByRole("dialog", { name: "Mobile Navigation Menu" })).toBeInTheDocument());

        // Close mobile menu
        const closeButton = screen.getByTestId("Close-Mobile-Menu");
        fireEvent.click(closeButton);

        // Wait for the mobile menu to disappear
        await waitFor(() =>
            expect(screen.queryByRole("dialog", { name: "Mobile Navigation Menu" })).not.toBeInTheDocument(),
        );
    });

    it("calls sign out and clears session storage when clicking 'Sign Out'", () => {
        sessionStorage.setItem("auth-token", "mock-token");
        sessionStorage.setItem("user", "mock-user");

        renderWithTheme(<Navbar />);

        // Sign out
        const signOutButton = screen.getByText("Sign Out");
        fireEvent.click(signOutButton);

        // Verify session storage is cleared
        expect(sessionStorage.getItem("auth-token")).toBeNull();
        expect(sessionStorage.getItem("user")).toBeNull();
    });

    it("renders correctly in mobile view", async () => {
        // Simulate mobile viewport
        Object.defineProperty(window, "innerWidth", { writable: true, configurable: true, value: 375 });
        window.dispatchEvent(new Event("resize"));

        sessionStorage.setItem("auth-token", "mock-token");

        const { asFragment } = renderWithTheme(<Navbar />);

        // Verify mobile elements using `data-testid`
        const hamburgerButton = screen.getByTestId("HamburgerMenu");
        expect(hamburgerButton).toBeInTheDocument();

        // Match snapshot
        expect(asFragment()).toMatchSnapshot();
    });
});
