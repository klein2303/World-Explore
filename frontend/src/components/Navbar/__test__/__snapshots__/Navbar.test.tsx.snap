// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar Component > renders correctly in mobile view 1`] = `
<DocumentFragment>
  <main>
    <nav
      aria-label="Main Navigation"
      class="_navbar_278c00 _light_278c00"
      role="navigation"
    >
      <svg
        aria-description="Open Mobile Menu"
        class="_hamburgmenu_278c00"
        data-testid="HamburgerMenu"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        tabindex="0"
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M1.5 3C1.22386 3 1 3.22386 1 3.5C1 3.77614 1.22386 4 1.5 4H13.5C13.7761 4 14 3.77614 14 3.5C14 3.22386 13.7761 3 13.5 3H1.5ZM1 7.5C1 7.22386 1.22386 7 1.5 7H13.5C13.7761 7 14 7.22386 14 7.5C14 7.77614 13.7761 8 13.5 8H1.5C1.22386 8 1 7.77614 1 7.5ZM1 11.5C1 11.2239 1.22386 11 1.5 11H13.5C13.7761 11 14 11.2239 14 11.5C14 11.7761 13.7761 12 13.5 12H1.5C1.22386 12 1 11.7761 1 11.5Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <a
        class="_navTitle_278c00"
        href="/"
      >
        <p>
          WorldExplore
        </p>
      </a>
      <div
        aria-label="Navigation Links"
        class="_navlinks_278c00"
        role="group"
      >
        <a
          class="_navlink_278c00"
          href="/ExploreCountries"
        >
          <svg
            aria-hidden="true"
            class="_search_278c00"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 32 32"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M 16 4 C 9.382813 4 4 9.382813 4 16 C 4 22.617188 9.382813 28 16 28 C 22.617188 28 28 22.617188 28 16 C 28 9.382813 22.617188 4 16 4 Z M 16 6 C 17.96875 6 19.796875 6.585938 21.34375 7.5625 L 20.53125 7.65625 L 20.71875 9.65625 L 19.65625 9.1875 L 18.78125 9.9375 L 18.9375 12 L 21.09375 11.3125 L 23.78125 12.1875 L 23.09375 13.4375 L 21.46875 12.4375 L 19.71875 12.6875 L 18 13.96875 L 17.03125 16.96875 L 18.96875 18.5625 C 18.96875 18.5625 20.957031 18.21875 21.0625 18.21875 C 21.167969 18.21875 21.90625 20.03125 21.90625 20.03125 L 20.3125 25.03125 C 19.011719 25.652344 17.542969 26 16 26 C 15.683594 26 15.371094 25.964844 15.0625 25.9375 L 13.96875 24.03125 L 15.03125 20.03125 L 11 17 L 7.28125 17 L 6.3125 15.03125 L 9 12.90625 L 13 11 L 12.40625 8.34375 L 14.125 7.96875 L 14.9375 9.09375 L 17.9375 8.53125 L 17.40625 6.21875 L 15.1875 6.0625 C 15.453125 6.042969 15.726563 6 16 6 Z M 14.875 6.0625 L 13.3125 6.71875 L 12.5625 6.59375 C 13.300781 6.324219 14.066406 6.152344 14.875 6.0625 Z M 6.0625 16.78125 L 7.03125 17.90625 L 7.03125 19.96875 L 8.90625 22.03125 L 10.0625 22.03125 L 12.90625 25.53125 C 9.132813 24.308594 6.378906 20.890625 6.0625 16.78125 Z"
            />
          </svg>
          <p>
            Explore Countries
          </p>
        </a>
        <a
          class="_navlink_278c00"
          href="/MyJournals"
        >
          <svg
            aria-hidden="true"
            class="_pin_278c00"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 10c0 6-8 12-8 12s-8-6-8-12a8 8 0 0 1 16 0Z"
            />
            <circle
              cx="12"
              cy="10"
              r="3"
            />
          </svg>
          <p>
            My Journals
          </p>
        </a>
        <a
          aria-label="Sign out"
          class="_navlink_278c00"
          href="/"
        >
          <svg
            aria-hidden="true"
            class="_profile_278c00"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5C14.2091 5 16 6.79086 16 9ZM14 9C14 10.1046 13.1046 11 12 11C10.8954 11 10 10.1046 10 9C10 7.89543 10.8954 7 12 7C13.1046 7 14 7.89543 14 9Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM3 12C3 14.0902 3.71255 16.014 4.90798 17.5417C6.55245 15.3889 9.14627 14 12.0645 14C14.9448 14 17.5092 15.3531 19.1565 17.4583C20.313 15.9443 21 14.0524 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12ZM12 21C9.84977 21 7.87565 20.2459 6.32767 18.9878C7.59352 17.1812 9.69106 16 12.0645 16C14.4084 16 16.4833 17.1521 17.7538 18.9209C16.1939 20.2191 14.1881 21 12 21Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <p>
            Sign Out
          </p>
        </a>
        <svg
          aria-label="Switch to dark mode"
          color="#424242"
          fill="none"
          height="25"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="cursor: pointer; transform: rotate(90deg); margin-left: 10px;"
          tabindex="0"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="circle-mask-7"
          >
            <rect
              fill="white"
              height="100%"
              width="100%"
              x="0"
              y="0"
            />
            <circle
              fill="black"
              r="9"
            />
          </mask>
          <circle
            cx="12"
            cy="12"
            fill="#424242"
            mask="url(#circle-mask-7)"
          />
          <g
            stroke="currentColor"
            style="opacity: 1;"
          >
            <line
              x1="12"
              x2="12"
              y1="1"
              y2="3"
            />
            <line
              x1="12"
              x2="12"
              y1="21"
              y2="23"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="4.22"
              y2="5.64"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="18.36"
              y2="19.78"
            />
            <line
              x1="1"
              x2="3"
              y1="12"
              y2="12"
            />
            <line
              x1="21"
              x2="23"
              y1="12"
              y2="12"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="19.78"
              y2="18.36"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="5.64"
              y2="4.22"
            />
          </g>
        </svg>
      </div>
    </nav>
  </main>
</DocumentFragment>
`;

exports[`Navbar Component > renders correctly when user is logged in and matches snapshot 1`] = `
<DocumentFragment>
  <main>
    <nav
      aria-label="Main Navigation"
      class="_navbar_278c00 _light_278c00"
      role="navigation"
    >
      <svg
        aria-description="Open Mobile Menu"
        class="_hamburgmenu_278c00"
        data-testid="HamburgerMenu"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        tabindex="0"
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M1.5 3C1.22386 3 1 3.22386 1 3.5C1 3.77614 1.22386 4 1.5 4H13.5C13.7761 4 14 3.77614 14 3.5C14 3.22386 13.7761 3 13.5 3H1.5ZM1 7.5C1 7.22386 1.22386 7 1.5 7H13.5C13.7761 7 14 7.22386 14 7.5C14 7.77614 13.7761 8 13.5 8H1.5C1.22386 8 1 7.77614 1 7.5ZM1 11.5C1 11.2239 1.22386 11 1.5 11H13.5C13.7761 11 14 11.2239 14 11.5C14 11.7761 13.7761 12 13.5 12H1.5C1.22386 12 1 11.7761 1 11.5Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <a
        class="_navTitle_278c00"
        href="/"
      >
        <p>
          WorldExplore
        </p>
      </a>
      <div
        aria-label="Navigation Links"
        class="_navlinks_278c00"
        role="group"
      >
        <a
          class="_navlink_278c00"
          href="/ExploreCountries"
        >
          <svg
            aria-hidden="true"
            class="_search_278c00"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 32 32"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M 16 4 C 9.382813 4 4 9.382813 4 16 C 4 22.617188 9.382813 28 16 28 C 22.617188 28 28 22.617188 28 16 C 28 9.382813 22.617188 4 16 4 Z M 16 6 C 17.96875 6 19.796875 6.585938 21.34375 7.5625 L 20.53125 7.65625 L 20.71875 9.65625 L 19.65625 9.1875 L 18.78125 9.9375 L 18.9375 12 L 21.09375 11.3125 L 23.78125 12.1875 L 23.09375 13.4375 L 21.46875 12.4375 L 19.71875 12.6875 L 18 13.96875 L 17.03125 16.96875 L 18.96875 18.5625 C 18.96875 18.5625 20.957031 18.21875 21.0625 18.21875 C 21.167969 18.21875 21.90625 20.03125 21.90625 20.03125 L 20.3125 25.03125 C 19.011719 25.652344 17.542969 26 16 26 C 15.683594 26 15.371094 25.964844 15.0625 25.9375 L 13.96875 24.03125 L 15.03125 20.03125 L 11 17 L 7.28125 17 L 6.3125 15.03125 L 9 12.90625 L 13 11 L 12.40625 8.34375 L 14.125 7.96875 L 14.9375 9.09375 L 17.9375 8.53125 L 17.40625 6.21875 L 15.1875 6.0625 C 15.453125 6.042969 15.726563 6 16 6 Z M 14.875 6.0625 L 13.3125 6.71875 L 12.5625 6.59375 C 13.300781 6.324219 14.066406 6.152344 14.875 6.0625 Z M 6.0625 16.78125 L 7.03125 17.90625 L 7.03125 19.96875 L 8.90625 22.03125 L 10.0625 22.03125 L 12.90625 25.53125 C 9.132813 24.308594 6.378906 20.890625 6.0625 16.78125 Z"
            />
          </svg>
          <p>
            Explore Countries
          </p>
        </a>
        <a
          class="_navlink_278c00"
          href="/MyJournals"
        >
          <svg
            aria-hidden="true"
            class="_pin_278c00"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 10c0 6-8 12-8 12s-8-6-8-12a8 8 0 0 1 16 0Z"
            />
            <circle
              cx="12"
              cy="10"
              r="3"
            />
          </svg>
          <p>
            My Journals
          </p>
        </a>
        <a
          aria-label="Sign out"
          class="_navlink_278c00"
          href="/"
        >
          <svg
            aria-hidden="true"
            class="_profile_278c00"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5C14.2091 5 16 6.79086 16 9ZM14 9C14 10.1046 13.1046 11 12 11C10.8954 11 10 10.1046 10 9C10 7.89543 10.8954 7 12 7C13.1046 7 14 7.89543 14 9Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM3 12C3 14.0902 3.71255 16.014 4.90798 17.5417C6.55245 15.3889 9.14627 14 12.0645 14C14.9448 14 17.5092 15.3531 19.1565 17.4583C20.313 15.9443 21 14.0524 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12ZM12 21C9.84977 21 7.87565 20.2459 6.32767 18.9878C7.59352 17.1812 9.69106 16 12.0645 16C14.4084 16 16.4833 17.1521 17.7538 18.9209C16.1939 20.2191 14.1881 21 12 21Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <p>
            Sign Out
          </p>
        </a>
        <svg
          aria-label="Switch to dark mode"
          color="#424242"
          fill="none"
          height="25"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="cursor: pointer; transform: rotate(90deg); margin-left: 10px;"
          tabindex="0"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="circle-mask-1"
          >
            <rect
              fill="white"
              height="100%"
              width="100%"
              x="0"
              y="0"
            />
            <circle
              fill="black"
              r="9"
            />
          </mask>
          <circle
            cx="12"
            cy="12"
            fill="#424242"
            mask="url(#circle-mask-1)"
          />
          <g
            stroke="currentColor"
            style="opacity: 1;"
          >
            <line
              x1="12"
              x2="12"
              y1="1"
              y2="3"
            />
            <line
              x1="12"
              x2="12"
              y1="21"
              y2="23"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="4.22"
              y2="5.64"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="18.36"
              y2="19.78"
            />
            <line
              x1="1"
              x2="3"
              y1="12"
              y2="12"
            />
            <line
              x1="21"
              x2="23"
              y1="12"
              y2="12"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="19.78"
              y2="18.36"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="5.64"
              y2="4.22"
            />
          </g>
        </svg>
      </div>
    </nav>
  </main>
</DocumentFragment>
`;

exports[`Navbar Component > renders correctly when user is not logged in and matches snapshot 1`] = `
<DocumentFragment>
  <main>
    <nav
      aria-label="Main Navigation"
      class="_notLoggedNavbar_278c00"
      role="navigation"
    >
      <svg
        aria-label="Open Mobile Menu"
        class="_hamburgmenu_278c00"
        fill="none"
        height="1em"
        id="mobile-menu-button"
        stroke="currentColor"
        stroke-width="0"
        tabindex="0"
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M1.5 3C1.22386 3 1 3.22386 1 3.5C1 3.77614 1.22386 4 1.5 4H13.5C13.7761 4 14 3.77614 14 3.5C14 3.22386 13.7761 3 13.5 3H1.5ZM1 7.5C1 7.22386 1.22386 7 1.5 7H13.5C13.7761 7 14 7.22386 14 7.5C14 7.77614 13.7761 8 13.5 8H1.5C1.22386 8 1 7.77614 1 7.5ZM1 11.5C1 11.2239 1.22386 11 1.5 11H13.5C13.7761 11 14 11.2239 14 11.5C14 11.7761 13.7761 12 13.5 12H1.5C1.22386 12 1 11.7761 1 11.5Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <a
        class="_navTitle_278c00"
        href="/"
      >
        <p>
          WorldExplore
        </p>
      </a>
      <div
        aria-label="Navigation Links"
        class="_navlinks_278c00"
        role="group"
      >
        <a
          class="_navlink_278c00"
          href="/Login"
        >
          <svg
            aria-label="Open Profile Menu"
            class="_profile_278c00"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5C14.2091 5 16 6.79086 16 9ZM14 9C14 10.1046 13.1046 11 12 11C10.8954 11 10 10.1046 10 9C10 7.89543 10.8954 7 12 7C13.1046 7 14 7.89543 14 9Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM3 12C3 14.0902 3.71255 16.014 4.90798 17.5417C6.55245 15.3889 9.14627 14 12.0645 14C14.9448 14 17.5092 15.3531 19.1565 17.4583C20.313 15.9443 21 14.0524 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12ZM12 21C9.84977 21 7.87565 20.2459 6.32767 18.9878C7.59352 17.1812 9.69106 16 12.0645 16C14.4084 16 16.4833 17.1521 17.7538 18.9209C16.1939 20.2191 14.1881 21 12 21Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <p>
            Log in
          </p>
        </a>
        <svg
          aria-label="Switch to dark mode"
          color="#424242"
          fill="none"
          height="25"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="cursor: pointer; transform: rotate(90deg); margin-left: 10px;"
          tabindex="0"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="circle-mask-2"
          >
            <rect
              fill="white"
              height="100%"
              width="100%"
              x="0"
              y="0"
            />
            <circle
              fill="black"
              r="9"
            />
          </mask>
          <circle
            cx="12"
            cy="12"
            fill="#424242"
            mask="url(#circle-mask-2)"
          />
          <g
            stroke="currentColor"
            style="opacity: 1;"
          >
            <line
              x1="12"
              x2="12"
              y1="1"
              y2="3"
            />
            <line
              x1="12"
              x2="12"
              y1="21"
              y2="23"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="4.22"
              y2="5.64"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="18.36"
              y2="19.78"
            />
            <line
              x1="1"
              x2="3"
              y1="12"
              y2="12"
            />
            <line
              x1="21"
              x2="23"
              y1="12"
              y2="12"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="19.78"
              y2="18.36"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="5.64"
              y2="4.22"
            />
          </g>
        </svg>
      </div>
    </nav>
  </main>
</DocumentFragment>
`;
